dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java

```diff
@@ -184,8 +187,47 @@ private String normalizePath(String basePath, String path) {
             return path.startsWith("/") ? basePath + path : basePath + "/" + path;
         }
 
-        public List<String> getEndpoints() {
-            return Ordering.natural().sortedCopy(endpoints);
+        public List<EndpointLogLine> getEndpointLogLines() {
+            return endpointLogLines;
+        }
+    }
+
+    private static class EndpointLogLine {
+        private final String httpMethod;
+        private final String basePath;
+        private final Class<?> klass;
+
+        public EndpointLogLine(String httpMethod, String basePath, Class<?> klass) {
+            this.basePath = basePath;
+            this.klass = klass;
+            this.httpMethod = httpMethod;
+        }
+
+        public String getHttpMethod() {
+            return httpMethod;
+        }
+
+        public String getBasePath() {
+            return basePath;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("    %-7s %s (%s)", httpMethod, basePath, klass.getCanonicalName());
+        }
+    }
+
+    private static class EndpointComparator implements Comparator<EndpointLogLine>, Serializable {
```

Comment:
So much boiler code for equals, hashcode. That's why I chose the
comparator, want me to switch anyway?
On Apr 20, 2015 9:31 PM, "Artem Prigoda" <notifications@github.com> wrote:

> In
> dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java
> <https://github.com/dropwizard/dropwizard/pull/1002#discussion_r28720803>:
>
> > +
> > +        public String getHttpMethod() {
> > +            return httpMethod;
> > +        }
> > +
> > +        public String getBasePath() {
> > +            return basePath;
> > +        }
> > +
> > +        @Override
> > +        public String toString() {
> > +            return String.format("    %-7s %s (%s)", httpMethod, basePath, klass.getCanonicalName());
> > +        }
> > +    }
> > +
> > +    private static class EndpointComparator implements Comparator<EndpointLogLine>, Serializable {
>
> Could we just make EndpointLogLine to implement
> Comparable<EndpointLogLine>?
> If it so, we won't need to specify a comparator explicitly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropwizard/dropwizard/pull/1002/files#r28720803>.
>